app/src/org/commcare/android/logging/XPathErrorEntry.java

```diff
@@ -0,0 +1,149 @@
+package org.commcare.android.logging;
+
+import android.support.v4.util.Pair;
+
+import org.commcare.android.javarosa.AndroidLogger;
+import org.commcare.android.util.SessionStateUninitException;
+import org.commcare.dalvik.application.CommCareApp;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.session.CommCareSession;
+import org.commcare.suite.model.Profile;
+import org.javarosa.core.log.LogEntry;
+import org.javarosa.core.model.utils.DateUtils;
+import org.javarosa.core.services.storage.IMetaData;
+import org.javarosa.core.services.storage.Persistable;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+/**
+ * Log entry for xpath errors, capturing the expression, session, and cc app version.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class XPathErrorEntry extends LogEntry implements Persistable, IMetaData {
```

Comment:
@phillipm Given that you're implementing `Persistable` and `IMetaData` here, this should extend `AndroidLogEntry` instead, which already does that work for you.